aten/src/ATen/native/quantized/cpu/qclamp.cpp

```diff
@@ -84,14 +86,18 @@ Tensor quantized_clamp_impl(
   Tensor qy;
   if (min && max) {
 #ifdef USE_PYTORCH_QNNPACK
-    if (at::globalContext().qEngine() == at::QEngine::QNNPACK && qx.scalar_type() == kQUInt8) {
+    if (at::globalContext().qEngine() == at::QEngine::QNNPACK &&
```

Comment:
qnnpack does not support the other two cases, can you keep the original error for qnnpack?

Reply:
Does the following look ok?

Reply:
maybe you can separate the checks for qnnpack and fbgemm to make it more clear..

Reply:
i.e. don't repeat USE_PYTORCH_QNNPACK